docs/sources/clients/promtail/stages/drop.md

```diff
@@ -0,0 +1,179 @@
+```
+
+Would drop any log line with the word `debug` in it.
+
+#### Regex match a source
+
+Given the pipeline:
+
+```yaml
+- json:
+    expressions:
+     level:
+     msg:
+- drop:
+    source:     "level"
+    expression: "(error|ERROR)"
+```
+
+Would drop both of these log lines:
+
+```
+{"time":"2019-01-01T01:00:00.000000001Z", "level": "error", "msg":"11.11.11.11 - "POST /loki/api/push/ HTTP/1.1" 200 932 "-" "Mozilla/5.0 (Windows; U; Windows NT 5.1; de; rv:1.9.1.7) Gecko/20091221 Firefox/3.5.7 GTB6"}
+{"time":"2019-01-01T01:00:00.000000001Z", "level": "ERROR", "msg":"11.11.11.11 - "POST /loki/api/push/ HTTP/1.1" 200 932 "-" "Mozilla/5.0 (Windows; U; Windows NT 5.1; de; rv:1.9.1.7) Gecko/20091221 Firefox/3.5.7 GTB6"}
+```
+
+#### Value match a source
+
+Given the pipeline:
+
+```yaml
+- json:
+    expressions:
+     level:
+     msg:
+- drop:
+    source: "level"
+    value:  "error"
+```
+
+Would drop this log line:
+
+```
+{"time":"2019-01-01T01:00:00.000000001Z", "level": "error", "msg":"11.11.11.11 - "POST /loki/api/push/ HTTP/1.1" 200 932 "-" "Mozilla/5.0 (Windows; U; Windows NT 5.1; de; rv:1.9.1.7) Gecko/20091221 Firefox/3.5.7 GTB6"}
+```
+
+#### Dropping old log lines
+
+*NOTE* For `older_than` to work, you must be using the [timestamp](timestamp.md) stage to set the timestamp from the ingested log line _before_ applying the `drop` stage.
+
+Given the pipeline:
+
+```yaml
+- json:
+    expressions:
+     time:
+     msg:
+- timestamp:
+    source: time
+    format: RFC3339
+- drop:
+    older_than: 24h
+```
+
+With a current ingestion time of 2020-08-12T12:00:00Z would drop this log line when read from a file:
+
+```
+{"time":"2020-08-11T11:00:00Z", "level": "error", "msg":"11.11.11.11 - "POST /loki/api/push/ HTTP/1.1" 200 932 "-" "Mozilla/5.0 (Windows; U; Windows NT 5.1; de; rv:1.9.1.7) Gecko/20091221 Firefox/3.5.7 GTB6"}
+```
+
+However it would _not_ drop this log line:
+
+```
+{"time":"2020-08-11T13:00:00Z", "level": "error", "msg":"11.11.11.11 - "POST /loki/api/push/ HTTP/1.1" 200 932 "-" "Mozilla/5.0 (Windows; U; Windows NT 5.1; de; rv:1.9.1.7) Gecko/20091221 Firefox/3.5.7 GTB6"}
+```
+
+In this example the current time is 2020-08-12T12:00:00Z and `older_than` is 24h, any log read from a file in which the timestamp is extracted from the log, 
+and that timestamp is older than 2020-08-11T12:00:00Z will be dropped.
+
+#### Dropping long log lines
+
+Given the pipeline:
+
+```yaml
+- drop:
+    longer_than: 8192
+```
+
+Would drop any log line longer than 8192 bytes, this is useful when Loki would reject a line for being too long.
+
+
+### Complex Drops
+
+Complex `drop` stage configurations specify multiple options in one stage or specify multiple drop stages
+
+#### Drop logs by regex AND length
+
+Given the pipeline:
+
+```yaml
+- drop:
+    expression: ".*debug.*"
+    longer_than: 1024
+```
+
+Would drop all logs that contain the word _debug_ *AND* are longer than 1024 bytes
+
+#### Drop logs by time OR length OR regex
```

Comment:
```suggestion
#### Drop logs by time or length or regex
```

Reply:
Do not use capitalization for emphasis. Per the [Documentation style guide](https://github.com/grafana/grafana/blob/master/contribute/style-guides/documentation-style-guide.md), use italics for emphasis.

Reply:
In this case the capitalization is less for emphasis and more to indicate that it's a binary operation, where say in a programming language you might write `if x > 1 && x < 5` it's common to express this as `if x >1 AND x < 5` mostly to disambiguate the logical operation from the actual word, (AND from and)